cms/djangoapps/contentstore/tasks.py

```diff
@@ -125,3 +152,222 @@ def push_course_update_task(course_key_string, course_subscription_id, course_di
+        return
+    if not has_course_author_access(user, courselike_key):
+        with respect_language(language):
+            self.status.fail(_(u'Permission denied'))
+        return
+
+    library = isinstance(courselike_key, LibraryLocator)
+    if library:
+        root_name = LIBRARY_ROOT
+        courselike_module = modulestore().get_library(courselike_key)
+        import_func = import_library_from_xml
+    else:
+        root_name = COURSE_ROOT
+        courselike_module = modulestore().get_course(courselike_key)
+        import_func = import_course_from_xml
+
+    # Locate the uploaded OLX archive (and download it from S3 if necessary)
+    # Do everything in a try-except block to make sure everything is properly cleaned up.
+    data_root = path(settings.GITHUB_REPO_ROOT)
+    subdir = base64.urlsafe_b64encode(repr(courselike_key))
+    course_dir = data_root / subdir
+    try:
+        self.status.set_state(u'Unpacking')
+
+        if not archive_name.endswith(u'.tar.gz'):
+            with respect_language(language):
+                self.status.fail(_(u'We only support uploading a .tar.gz file.'))
+                return
+
+        temp_filepath = course_dir / get_valid_filename(archive_name)
+        if not course_dir.isdir():  # pylint: disable=no-value-for-parameter
+            os.mkdir(course_dir)
+
+        LOGGER.debug(u'importing course to {0}'.format(temp_filepath))
+
+        # Copy the OLX archive from where it was uploaded to (S3, Swift, file system, etc.)
+        if not course_import_export_storage.exists(archive_path):
+            LOGGER.info(u'Course import %s: Uploaded file %s not found', courselike_key, archive_path)
+            with respect_language(language):
+                self.status.fail(_(u'Tar file not found'))
+            return
+        with course_import_export_storage.open(archive_path, 'rb') as source:
+            with open(temp_filepath, 'wb') as destination:
+                def read_chunk():
+                    """
+                    Read and return up to 1024 bytes from the source file.
+                    """
+                    return source.read(1024)
+                for chunk in iter(read_chunk, b''):
+                    destination.write(chunk)
+        LOGGER.info(u'Course import %s: Download from storage complete', courselike_key)
+        # Delete from source location
+        course_import_export_storage.delete(archive_path)
+
+        # If the course has an entrance exam then remove it and its corresponding milestone.
+        # current course state before import.
+        if root_name == COURSE_ROOT:
+            if courselike_module.entrance_exam_enabled:
+                fake_request = RequestFactory().get(u'/')
+                fake_request.user = user
+                from contentstore.views.entrance_exam import remove_entrance_exam_milestone_reference
+                # TODO: Is this really ok?  Seems dangerous for a live course
```

Comment:
It does get recreated at the end of the import process, I suspect it was just a hack to avoid refactoring the import code.  I've heard that at least MIT does a lot of content authoring directly in OLX for features that aren't supported in Studio yet, so it wouldn't surprise me if re-imports of live courses happen more often than we would advise.